c10/core/ScalarType.h

```diff
@@ -50,7 +52,9 @@ namespace c10 {
   _(c10::bits2x4, Bits2x4) /* 19 */                      \
   _(c10::bits4x2, Bits4x2) /* 20 */                      \
   _(c10::bits8, Bits8) /* 21 */                          \
-  _(c10::bits16, Bits16) /* 22 */
+  _(c10::bits16, Bits16) /* 22 */                        \
+  _(c10::Float8_e5m2, Float8_e5m2) /* 23 */              \
+  _(c10::Float8_e4m3, Float8_e4m3) /* 24 */
```

Comment:
@vkuzo, folks pointed out to me on Twitter https://twitter.com/ezyang/status/1665399428809629696 that there are multiple e4m3 variants; GraphCore has its own separate NaN variant. In JAX, these variants are distinguished, see https://github.com/jax-ml/ml_dtypes#specifications-of-implemented-floating-point-formats for their naming convention. Since I don't think we're expecting people to actually use these dtype names directly, it seems... OK to keep inline with the LLVM/JAX naming conventions?

Reply:
Done.